examples/switch_mode_persist.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+# Switch between modes without reallocating buffers.
+
+import time
+
+from picamera2 import Picamera2, Preview
+from picamera2.allocators import PersistentAllocator
+
+picam2 = Picamera2()
+picam2.allocator = PersistentAllocator()
+picam2.start_preview(Preview.QTGL)
+
+# Create preview configuration
+preview_config = picam2.create_preview_configuration()
+picam2.configure(preview_config)
+
+size = picam2.sensor_resolution
+# GPU won't digest images wider than 4096 on a Pi 4.
+if size[0] > 4096:
+    height = size[1] * 4096 // size[0]
+    height -= height % 2
+    size = (4096, height)
+
+# Create other configuration
+other_config = picam2.create_preview_configuration(main={"size": size}, buffer_count=2)
+other_config["use_case"] = "other"
```

Comment:
Ah, OK, so the buffers are accessed through the "use_case" name. I wonder about allowing `use_case="other"` in the `create_xxx_configuration` functions. (Though that does clutter them up even more, so I'm not sure...).

I always thought the "use_case" might be a useful concept, but in the event I never found a use for it, so I suppose that makes it quite a good solution here.

Reply:
I can add `use_case="xxx"` arguments to the `create_xxx_configuration` functions - it does add a little bit more clutter, but I think that does sound like a good idea, to make it easier to use